backend/battles/views.py

```diff
@@ -86,3 +88,33 @@ def get_context_data(self, **kwargs):  # noqa
             Q(user_creator=user) | Q(user_opponent=user), settled=False
         )
         return context
+
+
+class DetailBattleView(UserIsNotInThisBattleMixin, DetailView):
+    model = Battle
+    template_name = "battles/detail_battle.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        # Determine if you are opponent or creator
+        battle = context["battle"]
+        if self.request.user == battle.user_creator:
+            opponent = battle.user_opponent
+        else:
+            opponent = battle.user_creator
+
+        context["opponent"] = opponent.get_short_name
+        your_team = self.request.user.teams.get(battle=battle).team.all()
+
+        # Battle is settled
+        if battle.settled:
+            context["winner"] = battle.winner.get_short_name
+            opponent_team = opponent.teams.get(battle=battle).team.all()
+            context["pokemon"] = zip(your_team, opponent_team)
+
+            return context
+
+        # Battle is not settled
+        context["pokemon"] = zip(your_team, [0, 0, 0])
```

Comment:
How about you extract this to a helper?

Reply:
Done! It's a difficult code to organize but I hope it's ok

Reply:
I can't see the code (?), did you move it somewhere else?